riscv/src/reachability.rs

```diff
@@ -39,63 +82,39 @@ pub fn filter_reachable_from(
         } else if let Some(offset) = label_offsets.get(l) {
             let (referenced_labels_in_block, seen_labels_in_block) =
                 basic_block_references_starting_from(&statements[*offset..]);
-            assert!(!secondary_labels.contains(l));
-            secondary_labels.extend(seen_labels_in_block.iter());
-            secondary_labels.remove(l);
             processed_labels.extend(seen_labels_in_block);
             referenced_labels_in_block
         } else {
             eprintln!("The RISCV assembly code references an external routine / label that is not available:");
             eprintln!("{l}");
             panic!();
         };
-        for referenced in &new_references {
+        for referenced in new_references {
             if !processed_labels.contains(referenced) {
                 queued_labels.insert(referenced);
             }
         }
-        referenced_labels.extend(new_references);
     }
-    let code = processed_labels
-        .difference(&secondary_labels)
-        .flat_map(|l| {
-            let offset = *label_offsets.get(l).unwrap();
-            basic_block_code_starting_from(&statements[offset..])
```

Comment:
Why is the basic block stuff not needed anymore?

Reply:
It's done here now: https://github.com/powdr-org/powdr/pull/296/files#diff-a17aec442775d6ff04430fd25985a4c4caeb3e2cc84b334cdc2efd9661d2489dR35

Since we don't change the order of statements anymore, the strategy is just to first discover what labels are referenced and then go through the code from start to end and start including the statements as soon as we hit a referenced label until we have a statement that ends control-flow. Much, much easier.